lib/commands/implementations/gulag.js

```diff
@@ -25,65 +21,70 @@ function gulag(defaultBanTime) {
     if (parsedUsers.length > 5) {
       return new CommandOutput(null, 'Too many users to be sent to the gulag, max of 5.');
     }
-    gulagStarted = true;
+
     const listener = services.messageRelay.startListenerForChatMessages('gulag');
     if (listener === false) {
       return new CommandOutput(null, 'Something went wrong?? uhh. Restart me.');
     }
-    const votedMap = {};
-    const users = _.uniq(
-      parsedUsers.map((name) => {
-        if (name.toLowerCase() === 'random') {
-          return services.roleCache.getRecentRandomUsername();
-        }
-        return name;
-      }),
-    );
-    const userVotes = users.map((name) => ({ name, value: 0 }));
-    listener.on('message', (data) => {
-      if (votedMap[data.user]) {
-        return;
-      }
-      const message = data.message.trim();
-      if (isNormalInteger(message)) {
-        const int = parseInt(message, 10);
-        if (int >= 1 && int <= users.length) {
-          votedMap[data.user] = true;
-          userVotes[int - 1].value += 1;
-        }
-      }
-    });
-    setTimeout(() => {
-      gulagStarted = false;
+    listener.on('pollstop', (message) => {
+      const poll = JSON.parse(message);
+
       services.messageRelay.stopRelay('gulag');
-      services.messageRelay.sendOutputMessage('Total votes:');
-      userVotes.forEach((user) => {
-        services.messageRelay.sendOutputMessage(`${user.name} votes: ${user.value}`);
+      services.messageRelay.sendOutputMessage('GULAG has ended:');
+
+      const winnerVotes =
+        poll.totals[poll.totals.reduce((max, x, i, arr) => (x > arr[max] ? i : max), 0)];
```

Comment:
Is this just getting the highest number in the array? You can sort and get the item at the first index. We're dealing with tiny arrays here, so something suboptimal is acceptable.